pkg/objstore/providers/s3/bucket_client.go

```diff
@@ -39,8 +39,10 @@ func newS3Config(cfg Config) (s3.Config, error) {
 	}
 
 	bucketLookupType := s3.AutoLookup
-	if cfg.ForcePathStyle {
+	if cfg.ForcePathStyle || cfg.BucketLookupType == PathStyleLookup {
```

Comment:
I wonder if we also should add a deprecation log  warning when ForcePath  argument is used, so users can notice it and migrate to `BucketLookupType`

Wdyt?

Reply:
Yeah, that's a great idea. I have tried to add a warning on bucket client initialization. Maybe there is a better place for such logic?

Reply:
I think we could have also put it in validate, but that might have been shown even when no bucket is in use.

So that fine as it is. I noticed btw that we don't validate anything at present 😓  #3837